### DISCUSSION TITLE
HLS lab A chapter 7 Design Optimization lab1

### DISCUSSION THREAD

**Discussion by yuehfeng1114** [bol-edu/HLS-SOC-Discussions#27] [Q&A]
tutarial 是用2020.1版本的教學。但是之後的版本vitis會自動做了pipeline等等的優化，因此就算我們將pipeline手動關掉，出來的結果還是會被vitis優化過，因此沒辦法去分析不同做法的performance如何。想問說若是跑出來的結果無法解釋得通，能否只從tutarial中去分析它的結果呢?

**Comment by ciwade870307** [comment]
確實其他同學也有反應此問題，可能新版自動優化做太多了。與老師討論後有以下兩種做法 (1)根據你手動關掉的progma，看使否也能合理的分析各progma的影響 或是(2)直接使用tutorial中的結果來做分析。 只要以上有足夠的分析皆會給分
